tls/s2n_record_write.c

```diff
@@ -263,7 +278,11 @@ int s2n_record_writev(struct s2n_connection *conn, uint8_t content_type, const s
     /* Start the MAC with the sequence number */
     POSIX_GUARD(s2n_hmac_update(mac, sequence_number, S2N_TLS_SEQUENCE_NUM_LEN));
 
-    POSIX_GUARD(s2n_stuffer_resize_if_empty(&conn->out, S2N_LARGE_RECORD_LENGTH));
+    if (s2n_stuffer_is_freed(&conn->out)) {
```

Comment:
from offline convo: this isn't going to be a CPU regression because it does not increase the number of total resizes  per connection. If the application doesn't call release buffers, this will be the same number of resizes as s2n's previous behavior.